packages/cozy-client/src/models/konnectorFolder.js

```diff
@@ -0,0 +1,239 @@
+import { ensureMagicFolder, MAGIC_FOLDERS } from './folder'
+
+import CozyClient from '../CozyClient'
+import { getAccountName } from './account'
+
+import { getLocalizer } from './document/locales'
+
+const FILES_DOCTYPE = 'io.cozy.files'
+const PERMISSIONS_DOCTYPE = 'io.cozy.permissions'
+
+// Default name for base directory
+const DEFAULT_LOCALIZED_BASE_DIR = 'Administrative'
+
+/**
+ * Ensures the destination folder of a konnector exists and is initiated with proper permissions and references
+ *
+ * @param {CozyClient} client - CozyClent instance
+ * @param {Object} options - options object
+ * @param {import('../types').IOCozyKonnector} options.konnector - io.cozy.konnectors document
+ * @param {import('../types').IOCozyAccount} options.account - io.cozy.accounts document
+ * @param {String} options.lang - instance current language. ex: 'fr'
+ * @returns {Promise<import('../types').IOCozyFolder>}
+ */
+export const ensureKonnectorFolder = async (
+  client,
+  { konnector, account, lang }
+) => {
+  const permissions = client.collection(PERMISSIONS_DOCTYPE)
+  const files = client.collection(FILES_DOCTYPE)
+
+  const t = getLocalizer(lang)
+  const [adminFolder, photosFolder] = await Promise.all([
+    ensureMagicFolder(
+      client,
+      MAGIC_FOLDERS.ADMINISTRATIVE,
+      t('MagicFolders.administrative')
+    ),
+    ensureMagicFolder(client, MAGIC_FOLDERS.PHOTOS, t('MagicFolders.photos'))
+  ])
+
+  const path = buildFolderPath(konnector, account, {
+    // @ts-ignore
```

Comment:
why?

Reply:
Added 'attributes.path' in iocozyfolder type